contracts/ronin/validator/CandidateManager.sol

```diff
@@ -53,6 +54,46 @@ abstract contract CandidateManager is ICandidateManager, PercentageConsumer, Has
     require(!isValidatorCandidate(_consensusAddr), "CandidateManager: query for already existent candidate");
     require(_commissionRate <= _MAX_PERCENTAGE, "CandidateManager: invalid comission rate");
 
+    for (uint _i = 0; _i < _candidates.length; _i++) {
+      ValidatorCandidate storage existentInfo = _candidateInfo[_candidates[_i]];
```

Comment:
Should measure the gas used, IMO the apply-candidate tx should not be larger than 500k

Reply:
I have thought about it. There are two options:
1. Sacrifice execution gas one time per one validator. The later validators will need to spend more gas.
2. Need 4 more mappings on storage to support only this one function.

I'm more into option 1.

Reply:
In the worst case, there are 41 existent candidates, and `41 * 3 * 2100 ~= 260k` more gas is needed to query these values.

Reply:
I guess the number is acceptable